improve edit task and delete task use cases

hazemka opened the pull request:
- make the code more readability
- make the validation more specific
- add and improve test cases

hazemka committed:
improve edit task use case code
- make the code more readability
- make the validation more specific
- add and improve test cases

ibrahimdeeb0 commented on src/main/kotlin/logic/use_cases/tasks/EditTaskUseCse.kt:
Rename the class name, missing letter "a" in the keyword EditTaskUseCase

hazemka committed:
improve delete task use case
- make the code more readability
- make the validation more specific
- add new custom exception to handle with nullability
- improve delete task test cases

hazemka committed:
edit class name to correct typo
